### PR TITLE
MTDSA-2765 Integrated Components and added tests

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: test/v2/controllers/CharitableGivingControllerAmendSpec.scala [hunk 1]
```diff
@@ -16,58 +16,76 @@
...
-
-        MockedMtdIdLookupService.lookup(nino)
-          .returns(Future.successful(Left(UnauthorisedError)))
+    MockedMtdIdLookupService.lookup(nino).returns(Future.successful(Right("test-mtd-id")))
+    MockedEnrolmentsAuthService.authoriseUser()
+  }
 
-        private val result = target.amend(nino, taxYear)(fakeGetRequest)
-        status(result) shouldBe FORBIDDEN
+  val charitableGivingWithNonUKCharityDonations: String =
```
Comment (MarkAKelly) [inline test/v2/controllers/CharitableGivingControllerAmendSpec.scala]: If it helps, all of these are available in a test fixture.  Might tidy the tests up if we keep the data together.

File: it/v2/stubs/DesStub.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import support.WireMockMethods
+
+object DesStub extends WireMockMethods {
+
+  private def amendCharitableGivingUrl(nino: String, taxYear: String): String =
+    s"/income-tax/nino/$nino/income-source/charity/annual/$taxYear"
+
+  def amendSuccess(nino: String, taxYear: String): StubMapping = {
+    when(method = POST, uri = amendCharitableGivingUrl(nino, taxYear))
+      .thenReturn(status = OK, "{\"transactionReference\": \"12121\"}")
```
Comment (MarkAKelly) [inline it/v2/stubs/DesStub.scala]: I would maybe pull this response out of here and create it as a val or maybe put it in a test fixture.  Just a preference thing though.

File: test/v2/controllers/CharitableGivingControllerAmendSpec.scala (hunk 1 above)
Comment (narendravijayarao) [reply test/v2/controllers/CharitableGivingControllerAmendSpec.scala]: Is it going to be used any where else?

File: it/v2/stubs/DesStub.scala (hunk 2 above)
Comment (narendravijayarao) [reply it/v2/stubs/DesStub.scala]: I agree

Comment (MarkAKelly) [reply it/v2/stubs/DesStub.scala]: I would imagine it might possibly be used in the Retrieve endpoint.

**Pull request merged**